docs/src/prototypes/mentions/MentionsEditor.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+
+import { atMentionItems, AtMentionItem } from './dataMocks'
+import { insertTextAtCursorPosition } from './utils'
+import { PortalAtCursorPosition } from './PortalAtCursorPosition'
+
+export interface MentionsContainerProps {
+  items?: AtMentionItem[]
+  searchQuery?: string
+  open?: boolean
+  onInputKeyDown?: (e: React.KeyboardEvent) => void
+  onOpenChange?: (e: React.SyntheticEvent, data: { open?: boolean }) => void
+  onSearchQueryChange?: (e: React.SyntheticEvent, data: { searchQuery: string }) => void
+  onSelectedChange?: (e: React.SyntheticEvent, data: { searchQuery: string }) => void
+}
+
+interface MentionsEditorState {
+  shouldUpdate: boolean
+  open: boolean
+  searchQuery: string
+  selectedItem: string
+}
+
+interface MentionsEditorAction {
+  type: 'OPEN' | 'UPDATE_AND_CLOSE' | 'RESET_UPDATE_FLAG' | 'SET_SEARCH_QUERY'
+  args?: string
+}
+
+const editorStyle: React.CSSProperties = {
```

Comment:
_Nit_: we can use `createComponent()`

Reply:
I don't think I understand the reason behind it or the benefits 🤔